lib/with_win32.dart

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+import 'package:ffi/ffi.dart';
+import 'win32.dart' as win32;
+
+export 'win32.dart'
+    hide
+        CreateWindowEx,
+        DispatchMessage,
+        DrawText,
+        GetClientRect,
+        GetMessage,
+        RegisterClass,
+        TranslateMessage;
+
+T withMSG<T>(T Function(win32.MSG) func) {
+  final msg = win32.MSG.allocate();
+  T t;
+  try {
+    t = func(msg);
+  } finally {
+    free(msg.addressOf);
+  }
+  return t;
+}
+
+T withPAINTSTRUCT<T>(T Function(win32.PAINTSTRUCT) func) {
+  final ps = win32.PAINTSTRUCT.allocate();
+  T t;
+  try {
+    t = func(ps);
+  } finally {
+    free(ps.addressOf);
+  }
+  return t;
+}
+
+T withRECT<T>(T Function(win32.RECT) func) {
```

Comment:
This seems like it might get expensive, since `RECT`s are frequently passed from function to function.

Reply:
These `withXXX` functions are an attempt at making the lifetimes explicit, ala Rust and Java's try-with-resource. One of the wonderful things about Dart is the strength of it's GC implementation, as shown by Flutter's near blatant disregard for creating objects on each frame to describe the configuration of content on screen. So, I'm more worried about people `malloc`ing without matching `free` and thus slowly leaking an app to death, instead of chewing CPU.

IMNSHO, desktop class machines and GC are basically a solved problem for most non real time use cases. Look at C#. =)